app/src/components/task-completion/TaskCompletionBody.tsx

```diff
@@ -40,26 +42,67 @@ type TaskCompletionProps = {
 
 export default function TaskCompletion(props: TaskCompletionProps) {
   const { taskHeader, taskDescription } = props;
-  const [imageFile, setImageFile] = useState('');
+  const [imageFileSrc, setImageFileSrc] = useState('');
+  const [image, setImage] = useState<File | null>(null);
   const [isPopupActive, setIsPopupActive] = useState(false);
-  const hasImageBeenUploaded = imageFile !== '';
+  const hasImageBeenUploaded = imageFileSrc !== '';
+  const { id } = useParams<any>();
+
+  const alphanumericRegex = (str: string) => {
+    return str.replace(/[^a-zA-Z0-9]/g, '-');
+  };
+
+  const API_ROUTE = 'https://us-central1-scl-scavengerhunt.cloudfunctions.net/generateV4UploadSignedUrl';
+
+  const submitImage = async () => {
+    /**
+     *  1. create the filename
+     * 
+     *  2. post the filename to the backend
+     *    - receive specific upload url
+     * 
+     *  3. upload the file to that signed url
+     * 
+     *  4. (later) update our app DB that the task is completed
+     * 
+     */
+
+    const ext = image?.type.split('/')[1];
+    const contentType = image?.type;
+    let filename = `${id}-${alphanumericRegex(new Date().toISOString())}.${ext}`;
+
+    const signedUrl = await (await axios.get(API_ROUTE)).data.url;
```

Comment:
might be good to wrap either each call or this whole area of external calls in a try/catch